app/scripts/lib/channels/mixins/postmessage_receiver.js

```diff
@@ -90,9 +92,37 @@ define([
       errorIfNoResponse.call(this, outstanding);
     },
 
+    isEventFromExpectedOrigin: function (event) {
+      // `message` events that come from the Fx Desktop browser have an
+      // origin of the string constant 'null'. See
+      // https://developer.mozilla.org/docs/Web/API/Window/postMessage#Using_win.postMessage_in_extensions
+      // and
+      // https://bugzilla.mozilla.org/show_bug.cgi?id=1040257
+      //
+      // `message` events on Fx for iOS come from the page itself and
+      // should have the origin of the content server.
+      // See https://github.com/mozilla/firefox-ios/blob/66c8d746f0c8ca0f3c03122663205276b8ed85ab/FxAClient/Frontend/SignIn/FxAContentViewController.swift#L192
+      // `message` event from functional tests come from the page itself
+      // and should have the origin of the content server.
+
+      if (event.origin === 'null') {
+        return true;
+      }
```

Comment:
It may also be worth a more explicit comment on the logic here, to say "this can only have come from Fx Desktop, so it's safe to trust it"